conveyor/generator/generator/parser.go

```diff
@@ -0,0 +1,274 @@
+			Params: remap(p.sourceCode, methodType.Params),
+			Results: remap(p.sourceCode, methodType.Results),
+		}
+		switch {
+		case currentHandler.Name == "TID":
+		case currentHandler.Name == "Init":
+			p.parseInit(machine, currentHandler)
+		case strings.HasPrefix(currentHandler.Name, "State"):
+			p.parseState(machine, currentHandler)
+		case strings.HasPrefix(currentHandler.Name, "Transit"):
+			p.parseTransit(machine, currentHandler)
+		case strings.HasPrefix(currentHandler.Name, "Migrate"):
+			p.parseMigrate(machine, currentHandler)
+		case strings.HasPrefix(currentHandler.Name, "Error") && len(machine.States) == 0:
+			p.parseInitError(machine, currentHandler)
+		case strings.HasPrefix(currentHandler.Name, "Error"):
+			p.parseError(machine, currentHandler)
+		default:
+			log.Fatal("Unknokn handler: ", currentHandler.Name)
+		}
+	}
+	p.StateMachines = append(p.StateMachines, machine)
+	p.generator.stateMachines = append(p.generator.stateMachines, machine)
+}
+
+func (*Parser) parseInit(machine *stateMachine, h *handler) {
+	if len(h.Params) != 1 {
+		log.Fatal("Init must have only one parameter")
+	}
+	if len(h.Results) != 3 {
+		log.Fatal("Init should return three values")
+	}
+	if !strings.HasPrefix(h.Results[0], "*") {
+		log.Fatal("Returned payload should be a pointer")
+	}
+	if h.Results[1] != "common.ElState" {
+		log.Fatal("Returned state should be common.ElState")
+	}
+	if h.Results[2] != "error" {
+		log.Fatal("Returned error must be of type error")
+	}
+	if machine.Init != nil {
+		log.Fatal("Only one init handler for state machine")
+	}
+	machine.Init = &initHandler{
+		EventType: h.Params[0],
+		payloadType: h.Results[0],
+	}
+}
+
+func (*Parser) parseState(machine *stateMachine, h *handler) {
+	if len(h.Params) != 0 {
+		log.Fatal("State must not have any parameters")
+	}
+	if len(h.Results) != 1 || h.Results[0] != "common.ElState" {
+		log.Fatal("State should returns only common.ElState")
+	}
+	machine.States = append(machine.States, state{Name: h.Name})
+}
+
+func (*Parser) parseTransit(machine *stateMachine, h *handler) {
+	if len(h.Params) != 2 {
+		log.Fatal("Transit must have two parameters")
+	}
+	if h.Params[0] != machine.Init.EventType {
+		log.Fatal("Event should be of the same type with the event in the init")
+	}
+	if !strings.HasPrefix(h.Params[1], "*") {
+		log.Fatal("Payload must be a pointer")
+	}
+	if len(h.Results) != 3 {
+		log.Fatal("Transit should return three values")
+	}
+	if !strings.HasPrefix(h.Results[0], "*") {
+		log.Fatal("Returned payload should be a pointer")
+	}
+	if h.Results[1] != "common.ElState" {
+		log.Fatal("Returned state should be common.ElState")
+	}
+	if h.Results[2] != "error" {
+		log.Fatal("Returned error must be of type error")
+	}
+	if len(machine.States) < 1 {
+		log.Fatal("Declare state before handler")
+	}
+	if machine.States[len(machine.States)-1].Transit != nil {
+		log.Fatal("Only one transit handler for state")
+	}
+	machine.States[len(machine.States)-1].Transit = h
```

Comment:
we need test case which check that in handler interface all handlers go after 'State'

Reply:
We need a lot of negative tests